File: frontend/src/types/queue.ts
export interface QueueStatus {
  pending: number;
  processing: number;
  complete: number;
  error: number;
  workers_active: number;
}

export interface BatchStatus {
  batch_id: string;
  total: number;
  pending: number;
  processing: number;
  complete: number;
  error: number;
  submitted_at: string;
  jobs: JobSummary[];
}

export interface JobSummary {
  job_id: string;
  contract_address: string;
  status: 'pending' | 'processing' | 'complete' | 'error' | 'cancelled';
  risk_score: number | null;
  analysis_id: string;
}

export interface QueueSubmitResponse {
  batch_id: string | null;
  job_ids: string[];
  total: number;
}

File: backend/app/schemas/queue.py
"""Pydantic schemas for the queue REST endpoints."""

from __future__ import annotations

from typing import Optional

from pydantic import BaseModel, Field


class QueueSubmitRequest(BaseModel):
    """Payload for submitting one or more contract addresses."""

    addresses: list[str] = Field(
        ...,
        min_length=1,
        description="List of Solana contract addresses to analyse",
    )


class QueueSubmitResponse(BaseModel):
    """Response after a successful batch submission."""

    batch_id: str
    job_ids: list[str]
    total: int


class JobSummary(BaseModel):
    """Lightweight view of a single job's state."""

    job_id: str
    contract_address: str
    status: str
    worker_id: Optional[str] = None
    risk_score: Optional[float] = None
    error: Optional[str] = None


class QueueStatusResponse(BaseModel):
    """Aggregate queue health snapshot."""

    pending: int
    processing: int
    complete: int
    error: int
    workers_active: int


class BatchStatusResponse(BaseModel):
    """Detailed batch progress report."""

    batch_id: str
    total: int
    pending: int
    processing: int
    complete: int
    error: int
    jobs: list[JobSummary] = Field(default_factory=list)
